### ISSUE TITLE
`MD056` false positive for escaped pipe

### ISSUE THREAD

**Issue by FBruzzesi** [rvben/rumdl#163]
# Description

As per title, escaped pipe's in a table result in a false positive trigger for MD056.

**repro.md**

```md
# Example table

| Python type | Example  |
| ----------- | -------- |
| Union       | x \| y   |
| Union       | `x \| y` |
| Union       | x \\| y   |
| Union       | `x \\| y` |

```

Then running `rumdl`:

```terminal
> rumdl check repro.md
repro.md:5:1: [MD056] Table row has 3 cells, but expected 2 [*]
repro.md:6:1: [MD056] Table row has 3 cells, but expected 2 [*]
repro.md:7:1: [MD056] Table row has 3 cells, but expected 2 [*]
repro.md:8:1: [MD056] Table row has 3 cells, but expected 2 [*]

Issues: Found 4 issues in 1 file (16ms)
Run with `--fix` to automatically fix 4 of the 4 issues
```

Version: 

```terminal
> rumdl --version
rumdl 0.0.106
```

---

Rendering in github markdown:

| Python type | Example  |
| ----------- | -------- |
| Union       | x \| y   |
| Union       | `x \| y` |

---

**Update**: I checked the source and found this test:

https://github.com/rvben/rumdl/blob/7e52cf484b2b99a5723a061df42efb26d0763496/src/rules/md056_table_column_count.rs#L360-L373

yet updating with double `\\` doesn't seem to help

**Comment by rvben** [comment]
Hi @FBruzzesi, thanks for raising this issue and providing a clear example. There were some improvements to be made with regard to GFM spec compliance.
The improvements for this landed in 0.0.182 today. 🚀

**Comment by FBruzzesi** [comment]
Oh wow I just realized I was using a quite old version anyway! I am so sorry for reporting with such 🙈

Thanks for fixing it so quickly :)

**Comment by rvben** [comment]
A lot changed since 0.0.106! Haha. Enjoy using the latest version again! 😄
